broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -710,6 +663,57 @@ protected void handleFailure(final Exception failure) {
     onInstallFailure();
   }
 
+  private void addClosingStep(final String name, final AsyncClosable closable) {
+    closingSteps.add(new ClosingStep(name, closable));
+  }
+
+  private CompletableActorFuture<Void> closePartition() {
+    // this method may be called concurrently when the actor is closed
```

Comment:
I'm not sure how to prevent this situation. 

* first `closePartition()` is called by `transitionToLeader()`
* then `closePartition()` is called by `onActorClosing()`

What do you suggest in detail?

Reply:
Ok. I'll extend the code comment :+1: